### PR TITLE
[ticket/16771] Add MCP core events

### PR DESCRIPTION
**Pull request by rxu**
Checklist:

- [x] Correct branch: master for new features; 3.3.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/dev/master/development/coding_guidelines.html) and [3.3.x](https://area51.phpbb.com/docs/dev/3.3.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.3.x/development/git.html)

<a href="https://tracker.phpbb.com/browse/PHPBB3-16771">PHPBB3-16771</a>.

### CODE REVIEW

**Comment by rxu** [comment]
!set Event

File: phpBB/includes/mcp/mcp_topic.php [hunk 1]
```diff
@@ -420,7 +420,49 @@ function mcp_topic_view($id, $mode, $action)
 		'RETURN_FORUM'		=> sprintf($user->lang['RETURN_FORUM'], '<a href="' . append_sid("{$phpbb_root_path}viewforum.$phpEx", "f={$topic_info['forum_id']}&amp;start=$start") . '">', '</a>'),
 
 		'TOTAL_POSTS'		=> $user->lang('VIEW_TOPIC_POSTS', (int) $total),
-	));
+	];
+
+	/**
+	 * Event to modify the template data block for topic dsta output in the MCP
```
Comment (3D-I) [inline phpBB/includes/mcp/mcp_topic.php:426]:
```suggestion
	 * Event to modify the template data block for topic data output in the MCP
```

File: phpBB/includes/mcp/mcp_front.php [hunk 2]
```diff
@@ -119,11 +119,32 @@ function mcp_front_view($id, $mode, $action)
 						AND t.topic_id = p.topic_id
 						AND p.poster_id = u.user_id
 					ORDER BY p.post_time DESC, p.post_id DESC';
+
+				/**
+				* Alter posts data SQL query
```
Comment (3D-I) [inline phpBB/includes/mcp/mcp_front.php:124]: This doc-block does not follow the same (and suggested IMO) indentation of the others. See `*`'s

**Review by CHItA — APPROVED** [review]

File: phpBB/includes/mcp/mcp_front.php [hunk 3]
```diff
@@ -119,11 +119,32 @@ function mcp_front_view($id, $mode, $action)
 						AND t.topic_id = p.topic_id
 						AND p.poster_id = u.user_id
 					ORDER BY p.post_time DESC, p.post_id DESC';
+
+				/**
+				 * Alter posts data SQL query
+				 *
+				 * @event core.mcp_front_view_modify_posts_data_sql
+				 * @var	array	forum_list		List of forums that contain the posts
+				 * @var	array	forum_names		Associative array with forum_id as key and it's corresponding forum_name as value
```
Comment (marc1706) [inline phpBB/includes/mcp/mcp_front.php:128]:
```suggestion
				 * @var	array	forum_names		Associative array with forum_id as key and its corresponding forum_name as value
```

File: phpBB/includes/mcp/mcp_topic.php [hunk 4]
```diff
@@ -420,7 +420,49 @@ function mcp_topic_view($id, $mode, $action)
...
+	 * @var	string	mode					Mode of the MCP page we are displaying
+	 * @var	int		topic_id				The topic ID we are currently reviewing
+	 * @var	int		forum_id				The forum ID we are currently in
+	 * @var	bool	s_topic_icons			Flag indicating if split topic icon to be displayed
+	 * @var	int		start					Start item of this page
+	 * @var	string	subject					Subject of the topic to be split
+	 * @var	array	topic_info				Array with topic data
+	 * @var	int		to_forum_id				Forum id the topic is being moved to
+	 * @var	int		to_topic_id				Topic ID the topic is being merged with
+	 * @var	int		topic_row				Topic template data array
```
Comment (marc1706) [inline phpBB/includes/mcp/mcp_topic.php:442]: array

**Review by marc1706 — APPROVED** [review]

**Pull request merged**